developer_docs/translations.md

```diff
@@ -0,0 +1,53 @@
+## Translations Guidelines
+
+
+*How to contribute with Translations for p5 web editor*
+ 
+**General rules of thumb for translations**
+
+In order to simplify the translations process the following rules of thumb were used:
+
+**Technical Part**
+
+* There is only one file to translate all the texts in any specific language, which is located under the directory, in respective locale [subdirectory](https://github.com/processing/p5.js-web-editor/tree/develop/translations/locales)
+* The new language code must be added to [i18n.js](https://github.com/processing/p5.js-web-editor/blob/develop/client/i18n.js#L7)
+* New languages will need to be selected using a dropdown in Nav component, specifically in function [renderLanguageMenu.](https://github.com/processing/p5.js-web-editor/blob/develop/client/components/Nav.jsx#L550)
+ 
+**Translations**
+
+* Every component should introduce its own subset of keys inside a dictionary named after the component. 
+   For instance: If you want to translate AssetList.jsx you need to introduce the following namespace in translations.json :
+```
+ "AssetList": {
+    "Title": "p5.js Web Editor | My assets",
+    "ToggleOpenCloseARIA": "Toggle Open/Close Asset Options",
+    "OpenNewTab": "Open in New Tab",
+    "HeaderName": "Name",
+  }
+```
+* There are common texts that are present in Common namespace, try to check first if the exact label is already present there.
```

Comment:
We shouldn't use the term "namespace" here as that's a different concept in i18next and could be confusing confusing. Also, let's use the `code` style for "Common" so it's obvious it's a name in the code.

```suggestion
* There are common texts that are present with `Common` prefix, try to check first if the exact label is already present there.
```